ui/components/src/model/units/bytes.ts

```diff
@@ -31,16 +34,37 @@ export const BYTES_UNIT_CONFIG: Readonly<Record<BytesUnitKind, UnitConfig>> = {
 };
 
 // https://stackoverflow.com/questions/15900485/correct-way-to-convert-size-in-bytes-to-kb-mb-gb-in-javascript/18650828#18650828
-export function formatBytes(bytes: number, decimals = 2) {
+export function formatBytes(bytes: number, unitOptions: BytesUnitOptions) {
+  // default to full 'Bytes' formatting
+  const options = unitOptions.abbreviate === undefined ? { ...unitOptions, abbreviate: false } : unitOptions;
+
   if (bytes === 0) return '0 Bytes';
 
+  let decimals = options.decimal_places ?? DEFAULT_DECIMAL_PLACES;
+  // avoids minimumFractionDigits value is out of range error
+  if (decimals < 0) {
+    decimals = 0;
+  } else if (decimals > 100) {
+    decimals = 100;
+  }
+
+  if (options.abbreviate === false) {
+    const formatParams: Intl.NumberFormatOptions = {
+      style: 'decimal',
+      minimumFractionDigits: decimals,
+      maximumFractionDigits: decimals,
+      useGrouping: true,
+    };
+    const formatter = new Intl.NumberFormat('en-US', formatParams);
+    return `${formatter.format(bytes)} Bytes`;
+  }
+
   const k = 1024;
-  const dm = decimals < 0 ? 0 : decimals;
   const sizes = ['Bytes', 'KB', 'MB', 'GB', 'TB', 'PB', 'EB', 'ZB', 'YB'];
 
   // Math.max(0, ...) ensures that we don't return -1 as a value for the index.
   // Why? When the number of bytes are between -1 and 1, Math.floor(Math.log(bytes)/Math.log(1024)) returns -1.
   const i = Math.max(0, Math.floor(Math.log(bytes) / Math.log(k)));
 
-  return parseFloat((bytes / Math.pow(k, i)).toFixed(dm)) + ' ' + sizes[i];
+  return parseFloat((bytes / Math.pow(k, i)).toFixed(decimals)) + ' ' + sizes[i];
```

Comment:
Is it weird that above, we're using the `Intl.NumberFormat` API but down here, we're using `toFixed`? Would this benefit at all from using the same thing in either case?

Reply:
Yeah good point, I was a little scared to change this logic, but agreed it'll be simpler to update it above to just use `toFixed` as well